File: 1013-fibonacci-number/1013-fibonacci-number.cpp
class Solution {
public:
    int fib(int n) {
        long long memo[100];
        for(int i=0; i<100; i++){
            memo[i]=-1;
        }
        if(n<=1){
            return n;
        }
        if(memo[n]!=-1){
            return memo[n];
        }
        memo[n] = fib(n-1) + fib(n-2);
        return memo[n];
    }
};

File: 1013-fibonacci-number/1013-fibonacci-number.py
class Solution(object):
    def fib(self, n):
        """
        :type n: int
        :rtype: int
        """
        if n==0:
            return 0
        if n==1:
            return 1
        arr = [0] * (n+1)
        arr[0] = 0
        arr[1] = 1
        for i in range(2, n+1):
            arr[i] = arr[i-1] + arr[i-2]
        return arr[n]